Logger/Records.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace Logger
+{   //All the records are implemented implicitly so it wouldn't conflict with values with the same class
+    public record Book : IEntity
```

Comment:
```suggestion
    public record class Book(string Title, FullName Author, string ISBN) : RecordsEntity 
```
Define the members of book within the record class header, and then override their default implementations to null check.
Make sure to always specify what type of record it is after record.